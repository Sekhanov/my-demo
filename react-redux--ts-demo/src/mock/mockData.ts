import Person from "../data/Person";

export const persons: Person[] = [
    new Person("ivan", "petrov", 10),
    new Person('peter', 'ivanov', 20),
    new Person('Sergey', 'KHanov', 3)
];

export const mockStoreData = {
    persons: [
      {
        name: 'ivan',
        surname: 'petrov',
        age: 10
      },
      {
        name: 'peter',
        surname: 'ivanov',
        age: 20
      },
      {
        name: 'Sergey',
        surname: 'KHanov',
        age: 3
      }
    ]
  }