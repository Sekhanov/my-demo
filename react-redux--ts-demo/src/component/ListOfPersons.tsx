import React from 'react'
import Person from '../data/Person'
import IGlobalState from '../store/IGlobalState'
import { connect } from 'react-redux'
import { Persons } from '../mock/mockData';


interface IProps {
    persons : Person[];
}


export class PersonList extends React.Component<IProps> {
    

    render() {
        const personList = this.props.persons.map(person => 
            (<li key={person.age}>name: {person.name} surname: {person.surname}</li>)        
        )

        return ( 
            <ul>
                { personList }
            </ul>

        )
    }
}


const mapStateToProps = (state: IGlobalState) => ({
    persons: state.persons
})


export default connect(
    mapStateToProps
)(PersonList)