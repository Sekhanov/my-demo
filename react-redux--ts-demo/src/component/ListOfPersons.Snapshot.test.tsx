import React from 'react';
import renderer from 'react-test-renderer';
import { persons } from '../mock/mockData';
import { ListOfPersons } from './ListOfPersons';
import ListOfPersonsWithoutRedux from './ListOfPersonsWithoutRedux';





it("list of persons snapshot without redux", () => {
    const tree = renderer.create(
        <ListOfPersonsWithoutRedux
            persons={persons}
            someString={'some string from props'}
        ></ListOfPersonsWithoutRedux>
    )
    expect(tree).toMatchSnapshot();
})

it('list of person snapshot apart from redux, import { ListOfPersons }', () => {
    const tree = renderer.create(
        <ListOfPersons
            persons={persons}
            someString={'some string from props'}
        ></ListOfPersons>
    )
    expect(tree).toMatchSnapshot();
})





