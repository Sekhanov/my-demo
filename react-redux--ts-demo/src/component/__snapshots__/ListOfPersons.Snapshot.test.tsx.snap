// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list of person snapshot apart from redux, import { ListOfPersons } 1`] = `
<ul>
  <li>
    petrov
  </li>
  <li>
    ivanov
  </li>
  <li>
    KHanov
  </li>
  <p
    className="ss"
  >
    some string from props
  </p>
</ul>
`;

exports[`list of persons snapshot without redux 1`] = `
<ul>
  <li>
    name: 
    ivan
     surname: 
    petrov
  </li>
  <li>
    name: 
    peter
     surname: 
    ivanov
  </li>
  <li>
    name: 
    Sergey
     surname: 
    KHanov
  </li>
  <p
    className="ss"
  >
     
    some string from props
     
  </p>
</ul>
`;
